File: Javascript/MinimumAbsoluteDifferenceInBST.js
function TreeNode(val,left,right){
    this.val = val
    this.left = left
    this.right = right
}


var getMinimumDifference = function(root){
    const values = []
    function dfs(root){
        if (!root){
            return
        }

        dfs(root.left)
        values.push(root.val)
        dfs(root.right)
    }

    dfs(root)

    values.sort((a,b) => a-b)
    let minDiff = Infinity
    for (let i=0;i<values.length-1;i++){
        minDiff = Math.min(minDiff,Math.abs(values[i]-values[i+1]))
    }

    return minDiff
}

File: Python/MinimumAbsoluteDifferenceInBST.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def getMinimumDifference(root):

        def dfs(root,values):
            if root is None:
                return
            
            dfs(root.left,values)
            values.append(root.val)
            dfs(root.right,values)
            
        values = []
        dfs(root,values)

        values.sort()

        minDiff = float("inf")
        for i in range(len(values)-1):
            minDiff = min(minDiff,abs(values[i]-values[i+1]))

        return minDiff

